# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1238719850 14400
#      Thu Apr 02 20:50:50 2009 -0400
# Node ID 2d85197dbae4401c23fd9b8b3d3408b1652f6b97
# Parent  c79cf77061b78826198702803952d713ae09c3e4
__go_draw_axes__.m: Fix title placement for gnuplot 4.2.x.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2009-04-02  Ben Abbott <bpabbott@mac.com>
+
+	* plot/__go_draw_axes__.m: Fix title placement for gnuplot 4.2.x.
+
 2009-04-02  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/fsolve.m: Fix test.
 
 2009-03-30  Ben Abbott <bpabbott@mac.com>
 
 	* plot/__go_draw_axes__.m: New subfunction create_fontspec(). Allow
 	fontsize to be specified when the fontname is anonymous.
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -95,17 +95,17 @@ function __go_draw_axes__ (h, plot_strea
       if (isempty (t.string))
 	fputs (plot_stream, "unset title;\n");
       else
 	[tt, f, s] = __maybe_munge_text__ (enhanced, t, "string");
 	fontspec = create_fontspec (f, s, term);
 	fprintf (plot_stream, "set title \"%s\" %s %s",
 		 undo_string_escapes (tt), fontspec,
 		 __do_enhanced_option__ (enhanced, t));
-	if (nd == 3)
+	if (nd == 3 && __gnuplot_has_feature__ ("screen_coordinates_for_{lrtb}margin"))
 	  fprintf (plot_stream, " offset screen 0, screen %.3f;\n", pos(4)/5);
 	else
 	  fprintf (plot_stream, ";\n");
 	endif
       endif
     endif
 
     if (! isempty (axis_obj.xlabel))
